src-tauri/src/main.rs

```diff
@@ -195,15 +201,21 @@ pub(crate) async fn farm(
     // wait for the farming and plotting in the background
     tokio::spawn(async {
         tokio::select! {
-          res = plotting_instance.wait() => if let Err(error) = res {
-            return Err(anyhow!(error)) // TODO: connect this error to frontend, or log it
-          },
-          res = farming_instance.wait() => if let Err(error) = res {
-            return Err(anyhow!(error)) // TODO: connect this error to frontend, or log it
-          },
+            res = plotting_instance.wait() => if let Err(error) = res {
+                return Err(anyhow!(error))
+            },
+            res = farming_instance.wait() => if let Err(error) = res {
+                return Err(anyhow!(error))
+            },
         }
         Ok(())
     });
 
-    Ok(identity.public_key().to_bytes())
+    Ok(FarmerIdentity {
+        public_key: identity.public_key().to_bytes(),
+        mnemonic: Mnemonic::from_entropy(identity.entropy(), Language::English)
+            .unwrap()
+            .phrase()
+            .parse()?,
```

Comment:
Parsing `&str` to get `String` is non-idiomatic and I think in this case not needed:
```suggestion
        mnemonic: Mnemonic::from_entropy(identity.entropy(), Language::English)
            .unwrap()
            .into_phrase(),
```